app/controllers/shops_controller.rb

```diff
@@ -0,0 +1,64 @@
+class ShopsController < ApplicationController
+  def new
+    @company = Company.find(params[:company_id])
+    @address = Address.new()
+    @shop = Shop.new
+  end
+
+  def create
+    @company = Company.find(params[:company_id])
+    @address = Address.new(address_params)
+    @shop = Shop.new(shop_params.merge(company_id: @company.id))
+    @shop.address = @address
+    if @shop.save
+      redirect_to company_shop_path(@company, @shop)
+    else
+      render 'new'
+    end
+  end
+
+
+  def show
+    @company = Company.find(params[:company_id])
+    @shop = Shop.find(params[:id])
+  end
+
+  def edit
+    @shop = Shop.find(params[:id])
+    @company = Company.find(params[:company_id])
+  end
+
+  def update
+    @shop = Shop.find(params[:id])
```

Comment:
you should also find shop by company id not just shop id. as shop belongs to company